### ISSUE TITLE
CI: Recent runs seem to pick up the incorrect openblas?

### ISSUE THREAD

**Issue by seberg** [numpy/numpy#16146]
The `Linux_Python_38_32bit_full_with_asserts` azure run seems to fail a lot (or always) with the error:
```
AttributeError: /numpy/venv/lib/python3.8/site-packages/numpy/core/_multiarray_umath.cpython-38-i386-linux-gnu.so: undefined symbol: openblas_get_config
```
does that mean we are shipping a too small OpenBlas? Or, are we stripping the OpenBlas symbols and this one seemed unused and so get removed?

The windows builds is also failing due to mingw (often?), but I think there is nothing new about that.

**Comment by tylerjereddy** [comment]
Because 12 hours ago the manylinux repo [updated i686 images](https://github.com/pypa/manylinux/pull/565) to use a newer CentOS gcc toolchain, which means that the gfortran library that upstream OpenBLAS was built with is now too old for the manylinux 2010 image.

PEP 571 shows devtoolset-7: https://www.python.org/dev/peps/pep-0571/#docker-image

But I guess they are allowed to vary from the PEP, since they've bumped to devtoolset-8. That's a bit surprising to me I guess.

In the end, the effect is currently that this Azure CI Linux 32-bit entry builds  NumPy with no linear algebra backend, so naturally when the shared object file is probed for the OpenBLAS symbol it is not there.

**Comment by tylerjereddy** [comment]
Let me try pinning the image tag until a well thought out solution is agreed on.

**Comment by mattip** [comment]
Over at manylinux they suggested adding `yum install libgfortran4` until we update the openblas libs.

**Comment by tylerjereddy** [comment]
Of course updating the OpenBLAS binaries to build/link with a newer libgfortran could have other maintenance consequences for the broader ecosystem/project CIs.

I suppose it was going to happen eventually anyway.

**Comment by mattip** [comment]
manylinux added back the libgfortran in pypa/manylinux#571 needed by the older compiler, so it might be fine to try reverting gh-16147 at some point

**Comment by tylerjereddy** [comment]
Fixed by #16155 now that there's an upstream fix, thanks @mayeut.
